zrml/prediction-markets/src/lib.rs

```diff
@@ -1647,28 +1725,43 @@ mod pallet {
 
         /// Clears this market from being stored for automatic resolution.
-        fn clear_auto_resolve(market_id: &MarketIdOf<T>) -> DispatchResult {
+        fn clear_auto_resolve(market_id: &MarketIdOf<T>) -> Result<(u32, u32), DispatchError> {
             let market = T::MarketCommons::market(market_id)?;
-            if market.status == MarketStatus::Reported {
-                let report = market.report.ok_or(Error::<T>::MarketIsNotReported)?;
-                let dispute_duration_ends_at_block =
-                    report.at.saturating_add(market.deadlines.dispute_duration);
-                MarketIdsPerReportBlock::<T>::mutate(dispute_duration_ends_at_block, |ids| {
-                    remove_item::<MarketIdOf<T>, _>(ids, market_id);
-                });
-            }
-            if market.status == MarketStatus::Disputed {
-                let disputes = Disputes::<T>::get(market_id);
-                if let Some(last_dispute) = disputes.last() {
+            let (ids_len, disputes_len) = match market.status {
+                MarketStatus::Reported => {
+                    let report = market.report.ok_or(Error::<T>::MarketIsNotReported)?;
                     let dispute_duration_ends_at_block =
-                        last_dispute.at.saturating_add(market.deadlines.dispute_duration);
-                    MarketIdsPerDisputeBlock::<T>::mutate(dispute_duration_ends_at_block, |ids| {
-                        remove_item::<MarketIdOf<T>, _>(ids, market_id);
-                    });
+                        report.at.saturating_add(market.deadlines.dispute_duration);
+                    MarketIdsPerReportBlock::<T>::mutate(
+                        dispute_duration_ends_at_block,
+                        |ids| -> (u32, u32) {
+                            let ids_len = ids.len() as u32;
+                            remove_item::<MarketIdOf<T>, _>(ids, market_id);
+                            (ids_len, 0u32)
+                        },
+                    )
                 }
-            }
+                MarketStatus::Disputed => {
+                    let disputes = Disputes::<T>::get(market_id);
+                    if let Some(last_dispute) = disputes.last() {
+                        let dispute_duration_ends_at_block =
+                            last_dispute.at.saturating_add(market.deadlines.dispute_duration);
+                        MarketIdsPerDisputeBlock::<T>::mutate(
+                            dispute_duration_ends_at_block,
+                            |ids| -> (u32, u32) {
+                                let ids_len = ids.len() as u32;
+                                remove_item::<MarketIdOf<T>, _>(ids, market_id);
+                                (ids_len, disputes.len() as u32)
+                            },
+                        )
+                    } else {
+                        (0u32, disputes.len() as u32)
+                    }
+                }
+                _ => (0u32, 0u32),
```

Comment:
Sorry, but this is another edition of "I missed it in the first review". I think this branch **should** never happen, right? In that case, I would rather throw an error.

```suggestion
                _ => return Error::<T>::InvalidMarketStatus,
```

Reply:
In `admin_destroy_market` we also call this function `clear_auto_resolve` and I don't like to throw an error there, because it should be callable in any state.

Reply:
Oh, nice. :thumbsup: